packages/ui/src/components/Form/inputs.tsx

```diff
@@ -242,3 +242,101 @@ function ListItemInputRow<T>({
     </ListItem>
   );
 }
+
+/* A checkbox input that can be used to select multiple options */
+
+export type CheckboxInputOption<T> = {
+  title: string;
+  value: T;
+  description?: string;
+  disabled?: boolean;
+};
+
+export const CheckboxInput = <T,>({
+  option,
+  onChange,
+  checked,
+  ...props
+}: {
+  option: CheckboxInputOption<T>;
+  checked?: boolean;
+  onChange?: (value: T) => void;
+} & ComponentProps<typeof YStack>) => {
+  return (
+    <YStack {...props}>
+      <CheckboxInputRow
+        key={option.title}
+        option={option}
+        onPress={onChange}
+        checked={checked}
+      />
+    </YStack>
+  );
+};
+
+export function CheckboxInputRow<T>({
+  option,
+  onPress,
+  checked,
+}: {
+  option: CheckboxInputOption<T>;
+  onPress?: (value: T) => void;
+  checked?: boolean;
+} & Omit<ComponentProps<typeof RadioInputRowFrame>, 'onPress'>) {
+  const handlePress = useBoundHandler(option.value, onPress);
+  return (
+    <RadioInputRowFrame
+      onPress={option.disabled ? undefined : handlePress}
+      disabled={!!option.disabled}
+    >
+      <CheckboxControl
+        key={option.title}
+        disabled={option.disabled}
+        checked={checked}
+      />
+      <YStack gap="$l">
+        <LabelText size="$xl" color="$primaryText">
+          {option.title}
+        </LabelText>
+        {option.description ? (
+          <LabelText size="$m" color="$secondaryText">
+            {option.description}
+          </LabelText>
+        ) : null}
+      </YStack>
+    </RadioInputRowFrame>
+  );
+}
+
+const CheckboxControlFrame = styled(View, {
```

Comment:
`CheckboxControlFrame` and `CheckboxControl` look pretty similar to the radio ones. Could we avoid the duplication by allowing style override on the frame via props? Looks like border radius is the only difference?

Reply:
Thanks, took care of that!